### PR TITLE
Documentation fix for issue #15132 PrimaryHDU.fromstring() accepts bytes not bytearray for data parameter 

### LINKED ISSUE #15132: fits.PrimaryHDU.fromstring() errors when passing bytearray
### Description

I'm trying to create an HDU from a bytearray that includes the header and data. According to the documentation I can pass in a bytearray.  However when I try I get an error saying: TypeError: sequence item 0: expected str instance, bytearray found

It seems to be interpreting the bytearray as a bytearray but doesn't like it.

image

### Expected behavior

HDU created with the header and image data.

### How to Reproduce

1. FITS file loaded into a bytearray
2. Try and create a new PrimaryHDU with the astropy.io.fits.PrimaryHDU.fromstring() method passing in the bytearray and the ignore_missing_end=True option.

### Versions

Linux-5.15.0-1041-azure-x86_64-with-glibc2.35
Python 3.10.6 (main, May 29 2023, 11:10:38) [GCC 11.3.0]
astropy 5.3.1
Numpy 1.21.5
pyerfa 2.0.0.3
Scipy 1.9.1
Matplotlib 3.5.2

### PR DESCRIPTION
**Pull request by jaokeeffe**
### Description

Documentation fix for issue #15132 PrimaryHDU.fromstring() accepts bytes not bytearray for data parameter of PrimaryHDU class to reflect it's correct usage.

Fixes #15132

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/coordinates/15155.other.rst
Comment (pllim) [inline docs/changes/coordinates/15155.other.rst:1]: We do not usually add change log for doc fix. But let's see if @saimn agrees. 

If you have to remove the change log entry, you can either remove it and squash the commits, or drop that commit where you added it in the first place. FYI.

Comment (jaokeeffe) [reply docs/changes/coordinates/15155.other.rst:1]: No worries - I saw it failed the change log action so thought I should create the entry. All good.

Comment (saimn) [reply docs/changes/coordinates/15155.other.rst:1]: Right, changelog is not needed for that kind of change, but it doesn't harm so I will merge like that. Thanks for the PR!

**Review by saimn — APPROVED** [review]

**Comment by saimn** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
@saimn , please use the "squash and merge" option. Thanks!

File: docs/changes/coordinates/15155.other.rst
Comment (saimn) [reply docs/changes/coordinates/15155.other.rst:1]: Oops, changelog should be in the fits directory and with the `bugfix` type (not `coordinates/15155.other.rst`). So maybe we can just remove it instead.

Comment (saimn) [reply docs/changes/coordinates/15155.other.rst:1]: And it's possible to delete files with github now, so I just did that.

**Comment by pllim** [comment]
@saimn , since you said it used to work, maybe no need to manually backport to v5.0.x ?

**Comment by saimn** [comment]
Used to, but probably from a looong time ago ;). 
I would bet the problem is already present in 5.0, but it's not a big deal either.

**Comment by pllim** [comment]
Since this is a one-liner, I applied the backport directly onto v5.0.x in e96035b9eb7febeff987f02ef4d283c41e5bd016 . Thanks!

**Pull request merged**